src/spyglass/common/common_usage.py

```diff
@@ -272,6 +274,26 @@ def make(self, key):
             query.list_file_paths(paper_key) + restr_graph.file_paths
         )
 
+        # Check for linked nwb objects and add them to the export
+        exclude_files = []
```

Comment:
It seems like you're adding items to file_paths, just to remove them again with the index, is that right? I think I would structure this as on overwrite, rather than a gather + remove
```python
keep_links = set() 
for file in file_paths:
    if not (links := get_linked(...)):
         keep_files.add(file)
    logger.warning(...)
file_paths = {'file_path': link for link in keep_links}
```

`unique_dicts` is slow. Adding things to a set enforces uniqueness as you go

Reply:
Fair, I 'll put in a cleaner method along these lines